RevenueCatUI/Templates/Components/LocalizedPartials.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  LocalizedPartials.swift
+//
+//  Created by Josh Holtz on 10/27/24.
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+protocol LocalizedPartial {}
+
+struct LocalizedStates<T: LocalizedPartial> {
+
+    let selected: T?
+    let introOffer: T?
```

Comment:
I don't think these are the same type of states 🤔 `introOffer` is really only ever applicable to packages, but `selected` can be applied to things like checkboxes and radiobuttons (which we don't have _now_).

I think `selected` state should be part of a more general system that allows any component to change their looks based upon it. To me, it falls into the same category as `hovered` (for any component) or `pressed` (for buttons). `introOffer` is not the same thing to me.